packages/ui/src/models/visualization/flows/kamelet-schema.service.ts

```diff
@@ -0,0 +1,59 @@
+import { IKameletDefinition } from '../../kamelets-catalog';
+import { VisualComponentSchema } from '../base-visual-entity';
+import { KameletBindingStep } from '../../camel-entities/kamelet-binding-overrides';
+import { useCatalogStore } from '../../../store';
+import { CatalogKind } from '../../catalog-kind';
+import { JSONSchemaType } from 'ajv';
+
+export class KameletSchemaService {
+  static getVisualComponentSchema(stepModel: KameletBindingStep): VisualComponentSchema | undefined {
+    if (stepModel === undefined) {
+      return undefined;
+    }
+    const definition = KameletSchemaService.getKameletDefinition(stepModel);
+    return {
+      title: definition?.metadata.name || '',
+      schema: KameletSchemaService.getSchemaFromKameletDefinition(definition),
+      definition: stepModel?.ref?.properties || {},
+    };
+  }
+
+  static getSchemaFromKameletDefinition(definition: IKameletDefinition | undefined): JSONSchemaType<unknown> {
```

Comment:
[very-minor] Could we please make these methods private if they are not intended to be used outside of this class?